tldextract/tldextract.py

```diff
@@ -374,12 +374,13 @@ def main():
     parser.add_argument('-c', '--cache_file', help='use an alternate TLD definition file')
 
     args = parser.parse_args()
+    tld_extract = TLDExtract(include_psl_private_domains=True)
```

Comment:
Anybody who runs `tldextract --update` and later uses the same cache file will be forced to have public/private domains in their results, even if they only want public domains.

Ideally, we store metadata in the persisted file so results can be filtered at runtime.

For the 1.x series though, I think I'd be happy with `include_psl_private_domains` being configurable here.